alert-common/src/main/java/com/synopsys/integration/alert/common/enumeration/PermissionKeys.java

```diff
@@ -0,0 +1,46 @@
+/**
+ * alert-common
+ *
+ * Copyright (c) 2019 Synopsys, Inc.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.synopsys.integration.alert.common.enumeration;
+
+public enum PermissionKeys {
+    CONFIG_PROVIDER_BLACKDUCK("config.provider.blackduck"),
+    CONFIG_PROVIDER_POLARIS("config.provider.polaris"),
+    CONFIG_CHANNEL_EMAIL("config.channel.email"),
+    CONFIG_CHANNEL_HIPCHAT("config.channel.hipchat"),
+    CONFIG_CHANNEL_SLACK("config.channel.slack"),
```

Comment:
Are we planning on adding permissions specifically for each channel/provider? I'm not sure how this will be implemented, but is there a was we could make this dynamic?

Reply:
With permissions the UI needs to be changed to look at these permissions keys and the backend as well when displaying descriptors or enabling or disabling functionality in the UI.  Therefore you have to a have a fixed set of permission keys.  I suppose a descriptor could provide permission keys that it cares about to be a bit more dynamic, but audit right now isn't a descriptor and we want to have permission to perform actions with the audit screen.

Reply:
Sounds like a possible future improvement then.